components/micro-gateway-core/src/main/ballerina/src/gateway/upload_files_timer.bal

```diff
@@ -1,92 +1,109 @@
 
-// import ballerina/io;
-// import ballerina/internal;
-// import ballerina/task;
-// import ballerina/math;
-// import ballerina/runtime;
-// import ballerina/log;
+import ballerina/task;
+import ballerina/file;
+import ballerina/http;
 
-// string uploadingUrl = "";
-// string analyticsUsername = "";
-// string analyticsPassword = "";
+string uploadingUrl = "";
+string analyticsUsername = "";
+string analyticsPassword = "";
 
 
-// function searchFilesToUpload() returns (error?) {
-//     int cnt = 0;
-//     string fileLocation = retrieveConfig(API_USAGE_PATH, API_USAGE_DIR);
-//     internal:Path path = new(fileLocation);
+function searchFilesToUpload() returns (error?) {
+    int cnt = 0;
+    string fileLocation = retrieveConfig(API_USAGE_PATH, API_USAGE_DIR);
 
-//     if (!path.exists()) {
-//         printDebug(KEY_UPLOAD_TASK, "Usage data directory not found");
-//         return ();
-//     }
+    if (!file:exists(fileLocation)) {
+        printDebug(KEY_UPLOAD_TASK, "Usage data directory not found");
+        return ();
+    }
 
-//     internal:Path[]|error pathList = path.list();
+    file:FileInfo[]|error pathList = file:readDir(fileLocation);
 
-//     if (pathList is error) {
-//         printError(KEY_UPLOAD_TASK, "Error occured in getting path lists");
-//         return pathList;
-//     } else {
-//         foreach var pathEntry in pathList {
-//             string fileName = pathEntry.getName();
-//             if (fileName.contains(ZIP_EXTENSION)) {
-//                 http:Response response = multipartSender(fileLocation + PATH_SEPERATOR, pathEntry.getName(),
-//                     analyticsUsername, analyticsPassword);
-//                 if (response.statusCode == 201) {
-//                     printInfo(KEY_UPLOAD_TASK, "Successfully uploaded the file: " + fileName);
-//                     var result = pathEntry.delete();
-//                 } else {
-//                     printError(KEY_UPLOAD_TASK, "Error occurred while uploading the file");
-//                 }
-//                 cnt=cnt +1;
-//             }
-//         }
-//         if (cnt == 0) {
-//             error er = error("No files present to upload.");
-//             return er;
-//         } else {
-//             return ();
-//         }
-//     }
-// }
+    if (pathList is error) {
+        printError(KEY_UPLOAD_TASK, "Error occured in getting path lists");
+        return pathList;
+    } else {
+        foreach var pathEntry in pathList {
+            string fileName = pathEntry.getName();
+            if (contains(fileName, ZIP_EXTENSION)) {
+                http:Response response = multipartSender(fileLocation + PATH_SEPERATOR, pathEntry.getName(),
+                    analyticsUsername, analyticsPassword);
+                if (response.statusCode == 201) {
+                    printInfo(KEY_UPLOAD_TASK, "Successfully uploaded the file: " + fileName);
+                    var result = file:remove(fileLocation + "/" + fileName);
```

Comment:
Check the forward slash will work on windows as well. Lets use the path seperator